modules/cudacodec/src/video_reader.cpp

```diff
@@ -395,14 +417,26 @@ Ptr<VideoReader> cv::cudacodec::createVideoReader(const String& filename, const
     }
 
     return makePtr<VideoReaderImpl>(videoSource, params.minNumDecodeSurfaces, params.allowFrameDrop, params.udpSource, params.targetSz,
-        params.srcRoi, params.targetRoi);
+        params.srcRoi, params.targetRoi, params.enableHistogramOutput);
 }
 
 Ptr<VideoReader> cv::cudacodec::createVideoReader(const Ptr<RawVideoSource>& source, const VideoReaderInitParams params)
 {
     Ptr<VideoSource> videoSource(new RawVideoSourceWrapper(source, params.rawMode));
     return makePtr<VideoReaderImpl>(videoSource, params.minNumDecodeSurfaces, params.allowFrameDrop, params.udpSource, params.targetSz,
-        params.srcRoi, params.targetRoi);
+        params.srcRoi, params.targetRoi, params.enableHistogramOutput);
+}
+
+void cv::cudacodec::MapHist(const GpuMat& hist, Mat& histFull) {
+    Mat histHost; hist.download(histHost);
+    histFull.create(histHost.size(), histHost.type());
+    histFull = 0;
+    const float scale = 255.0f / 219.0f;
+    const int offset = 16;
+    for (int iScaled = 0; iScaled < histHost.cols; iScaled++) {
+        const int iHistFull = std::min(std::max(0, static_cast<int>(std::round((iScaled - offset) * scale))), static_cast<int>(histFull.total()) - 1);
+        histFull.at<int>(iHistFull) += histHost.at<int>(iScaled);
+    }
```

Comment:
Why do we need the function in API. Looks like the same result may be achieved with convertTo(): https://docs.opencv.org/4.x/d0/d60/classcv_1_1cuda_1_1GpuMat.html#a656ac1f6a1426527f838c19e2d677dc0. It's done on GPU and could be a part of user code, if it's required. You just need to mention the conversion parameters in  `nextFrame`.

Reply:
Hi @asmorkalov, I'm not sure I understand.

If `cudacodec::FormatInfo::videoFullRangeFlag == false` the  hardware decoder generated histogram will have bins from 16-219 as the **encoded** luma component can only take these values.  When decoded to a gray image `img`, `img` will have a range of 0-255, therefore the histogram bins need to be mapped to 0-255 to be consistent with the result of calling `calcHist` on `img`.  Is there a way to use an existing function to perform this mapping?

Reply:
My fault, I miss-interpreted your code. Please ignore my proposal.